sendbuffer.go

```diff
@@ -60,16 +60,19 @@ func (buf *sendBuffer) sendLoop(w io.Writer) {
 		buf.closed.Done()
 	}()
 
-	closeTimer := time.NewTimer(largeTimeout)
+	closeTimedOut := make(chan interface{})
 	signalClose := func() {
-		buf.muClosing.Lock()
-		buf.closing = true
-		close(buf.in)
-		buf.muClosing.Unlock()
-		if !closeTimer.Stop() {
-			<-closeTimer.C
-		}
-		closeTimer.Reset(closeTimeout)
+		go func() {
+			buf.muClosing.Lock()
+			alreadyClosed := buf.closing
+			buf.closing = true
+			close(buf.in)
```

Comment:
If `alreadyClosed` can be true, does it mean `buf.in` can be closed more than once, which could cause a panic?

Reply:
`sync.Once` seems a good fit

Reply:
Good idea, done!